default-plugins/status-bar/src/tip/cache.rs

```diff
@@ -62,29 +65,79 @@ impl LocalCache {
                 file.read_to_string(&mut json_cache)
                     .map_err(LocalCacheError::Io)?;
 
-                let metadata = LocalCache::from_json(&json_cache)?;
-                Ok(LocalCache { path, metadata })
+                let res = LocalCache::from_json(&json_cache);
+                match res {
+                    Ok(metadata) => Ok(LocalCache { path, metadata }),
+                    Err(e) => {
+                        // JSON cache file is corrupted, or using a bad schema. Try to remove it
+                        std::fs::remove_file(&path)?;
+                        Err(e)
+                    },
+                }
             },
             Err(e) => Err(LocalCacheError::IoPath(e, path)),
         }
     }
 
-    pub fn flush(&mut self) -> Result<(), LocalCacheError> {
+    pub fn new(path: PathBuf) -> LocalCache {
+        let res = LocalCache::from_file(path.clone());
+        match res {
+            Ok(cache) => cache,
+            Err(e) => {
+                eprintln!(
+                    "Error loading status-bar cache from {:?}, error: {:?}",
+                    path, e
+                );
+                LocalCache {
+                    path,
+                    metadata: Metadata {
+                        zellij_version: get_zellij_version(),
+                        cached_data: HashMap::new(),
+                    },
+                }
+            },
+        }
+    }
+
+    fn safe_parent(p: &path::Path) -> Option<&path::Path> {
+        match p.parent() {
+            None => None,
+            Some(x) if x.as_os_str().is_empty() => Some(path::Path::new(".")),
+            x => x,
+        }
+    }
+
+    fn flush(&mut self) -> Result<(), LocalCacheError> {
         match serde_json::to_string(&self.metadata) {
             Ok(json_cache) => {
-                let mut file = File::create(self.path.as_path())
-                    .map_err(|e| LocalCacheError::IoPath(e, self.path.clone()))?;
+                let mut file = NamedTempFile::new_in(
+                    LocalCache::safe_parent(self.path.as_path()).ok_or_else(|| {
+                        LocalCacheError::IoPath(
+                            io::Error::new(
+                                io::ErrorKind::Other,
+                                "Could not get a parent path for tips cache",
+                            ),
+                            self.path.clone(),
+                        )
+                    })?,
+                )
+                .map_err(LocalCacheError::Io)?;
                 file.write_all(json_cache.as_bytes())
                     .map_err(LocalCacheError::Io)?;
+                file.persist(self.path.as_path())
```

Comment:
Hi @rdeaton, Thank you for contributing!

When I run your code in my environment, I keep getting the following error.
```
DEBUG  |/home/meow/Projects/zelli| 2022-12-20 21:27:49.223 [id: 1     ] Error flushing local cache to disk: Persist(PersistError { error: Os { code: 20, kind: AlreadyExists, message: "File exists" }, file: NamedTempFile("/tmp/.tmpKk1Sqm") }) 
```

Could you check re-again?